api/middleware/auth.ts

```diff
@@ -0,0 +1,39 @@
+import axios from 'axios'
+import * as passport from 'passport'
+import * as auth from '../../api/auth'
+import * as serviceTokenMiddleware from '../../api/auth/serviceToken'
+import * as log4jui from '../lib/log4jui'
+
+const logger = log4jui.getLogger('auth')
+
+export function validRoles(roles) {
+    //return roles.indexOf(config.juiJudgeRole) > -1 || roles.indexOf(config.juiPanelMember) > -1
+    return true //disabled role restriction for XUI
+}
+
+export default async (req, res, next) => {
+    if (req.isAuthenticated()) {
+
+        const userDetails = req.session.passport.user
+        const roles = userDetails.userinfo.roles
+
+        if (!validRoles(roles)) {
+            logger.warn('User role does not allow login')
+            return await auth.doLogout(req, res, 401)
+        }
+
+        logger.info('Auth token: ' + `Bearer ${userDetails.tokenset.access_token}`)
+
+        axios.defaults.headers.common.Authorization = `Bearer ${userDetails.tokenset.access_token}`
```

Comment:
Again, a security flaw to have the bearer token on axios instance, please refactor

Reply:
this will be a different PR as discussed with @dyanesh